pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/main/PipelineFactory.java

```diff
@@ -40,6 +43,65 @@
+
+    /**
+     * create an AnnotatorService with the given view names in the argument. The names are supposed be strings,
+     * separated by space.
+     *
+     * @return AnnotatorService with specified NLP components
+     * @throws IOException
+     * @throws AnnotatorException
+     */
+    public static BasicAnnotatorService buildPipeline(String... views) throws IOException,
+            AnnotatorException {
+        List<String> allViewNames = ViewNames.getAllViewNames();
+        Map<String, String> nonDefaultValues = new HashMap<>();
+        for(String vu : views) {
+            if( allViewNames.contains(vu) ) {
+                switch (vu) {
+                    case ViewNames.POS:
+                        nonDefaultValues.put(PipelineConfigurator.USE_POS.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.LEMMA:
+                        nonDefaultValues.put(PipelineConfigurator.USE_LEMMA.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.NER_CONLL:
+                        nonDefaultValues.put(PipelineConfigurator.USE_NER_CONLL.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.NER_ONTONOTES:
+                        nonDefaultValues.put(PipelineConfigurator.USE_NER_ONTONOTES.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.QUANTITIES:
+                        nonDefaultValues.put(PipelineConfigurator.USE_QUANTIFIER.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.SHALLOW_PARSE:
+                        nonDefaultValues.put(PipelineConfigurator.USE_SHALLOW_PARSE.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.SRL_VERB:
+                        nonDefaultValues.put(PipelineConfigurator.USE_SRL_VERB.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.DEPENDENCY_STANFORD:
+                        nonDefaultValues.put(PipelineConfigurator.USE_DEP.key, Configurator.TRUE);
+                        break;
+                    case ViewNames.PARSE_STANFORD:
+                        nonDefaultValues.put(PipelineConfigurator.USE_STANFORD_PARSE.key, Configurator.TRUE);
+                        break;
+                    default:
+                        logger.warn("View name is not supported yet. Look into the readme of the pipeline to see the list of valid annotators. ");
+                }
+            }
+            else {
+                throw new IllegalArgumentException("The view name " + vu + " is not a valid view name. " +
+                        "The possible view names are static members of the class `ViewName`. ");
+            }
+        }
+        // using the default settings and changing the views
+        ResourceManager fullRm = (new PipelineConfigurator()).getConfig(nonDefaultValues);
```

Comment:
OK -- please add a comment in README that this is the behavior if default constructor is used. 
Am OK with updating the version as Bhargav suggested.